File: app/src/main/java/kr/zagros/shwan/moviemvvm/views/CostumView/LimitedChatTextView.java
package kr.zagros.shwan.moviemvvm.views.CostumView;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.util.Log;



public class LimitedChatTextView extends AppCompatTextView {
    private static final String TAG = "LimitedChatTextView";

    private int maxTextChar;

    public LimitedChatTextView(Context context) {
        super(context);
    }

    public LimitedChatTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public LimitedChatTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public int getMaxTextChar() {
        return maxTextChar;
    }

    public void setMaxTextChar(int maxTextChar) {
        this.maxTextChar = maxTextChar;
    }

    @Override
    public void setText(CharSequence text, BufferType type) {
        CharSequence limitedText = text;
        if(maxTextChar<text.length())  limitedText = text.toString().substring(0, maxTextChar -1)+"...";
        Log.d(TAG, "setText: "+limitedText);
        super.setText(limitedText, type);
    }

}

File: app/src/main/java/kr/zagros/shwan/moviemvvm/views/CostumView/LimitedChatTextView.kt
package kr.zagros.shwan.moviemvvm.views.CostumView

import android.content.Context
import androidx.appcompat.widget.AppCompatTextView
import android.util.AttributeSet
import android.util.Log


class LimitedChatTextView : AppCompatTextView {

    var maxTextChar: Int = 0

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun setText(text: CharSequence, type: BufferType) {
        var limitedText = text
        if (maxTextChar < text.length) limitedText = text.toString().substring(0, maxTextChar - 1) + "..."
        Log.d(TAG, "setText: $limitedText")
        super.setText(limitedText, type)
    }

    companion object {
        private val TAG = "LimitedChatTextView"
    }

}
